### DISCUSSION TITLE
C# Azure Function passing object OrchestrationTrigger no longer working after update

### DISCUSSION THREAD

**Discussion by aardrasystems** [Azure/azure-functions-durable-extension#2418] [Q&A]
My Azure Function app was originally using Version 3, and everything was working well. Recently, I updated to Azure Function App V4, and I have encountered some issues with my durable function execution. I am not sure if this is related to the Azure Function App upgrade or some changes in the durable nugget package version. Originally, when using Function 3, I was using version 1.8.3, but now I am using the most recent version.

The issue is that when I start the durable function, I am no longer able to pass an object when triggering the OrchestrationTrigger, as I did previously.

Please see my code example below:

```
public static async Task<HttpResponseMessage> HttpStart(
    [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")] HttpRequestMessage req,
    [DurableClient] IDurableOrchestrationClient starter,
    ILogger log)
{
    // Function input comes from the request content.
    log.LogInformation("Started Device Control Checker");
    dynamic eventData = await req.Content.ReadAsAsync<object>();
    message_device_checker_obj message_control_durable = new message_device_checker_obj();
    string instanceId = await starter.StartNewAsync("devicecontrol_validation", eventData.message_control);
    return starter.CreateCheckStatusResponse(req, instanceId);
}
```

It seems that with StartNewAsync, I am no longer able to pass an object but just a simple string.

Can anyone advise me on how I can change this so that my object can be passed into the durable function?

I am then reading the object in the code below, but it's coming as NULL.

   ```
     public static async Task<List<string>> RunOrchestrator(
            [OrchestrationTrigger] IDurableOrchestrationContext context,ILogger log)
        {

            log.LogInformation($"Device Control Control Started--->");
            Iot_data_state iotstatedata = new Iot_data_state();            
            var outputs = new List<string>();
            message_device_checker_obj inputmessage = new message_device_checker_obj();
            outputs.Add("Stating Durable");

            iotstatedata.NextState = "int_zone";
            iotstatedata.device_control = context.GetInput<message_device_checker_obj>();            
            iotstatedata.termination_counter = 0;
            iotstatedata.sleepduration = 0;

            while (iotstatedata.NextState != "finished")
            {
                iotstatedata = await context.CallActivityAsync<Iot_data_state>("device_control_process", iotstatedata);
                outputs.Add(iotstatedata.NextState + " " + iotstatedata.now_time);

                if (iotstatedata.sleepduration > 0)
                {
                    DateTime deadline = context.CurrentUtcDateTime.Add(TimeSpan.FromSeconds(iotstatedata.sleepduration));
                    await context.CreateTimer(deadline, CancellationToken.None);
                }
            }
            return outputs;
        }

```

**Comment by davidmrdavid** [comment]
@aardrasystems: At a glance, I would suspect you're calling onto a different overload of the `StartNewAsync` method. See here for the overloads: https://github.com/Azure/azure-functions-durable-extension/blob/5cf4b110252e24ad6d00d1b2cca831d1a2e15c96/src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs#L1034-L1044

I suspect that now that you're using the new version, you'll need to call it like this: `string instanceId = await starter.StartNewAsync<object>("devicecontrol_validation", eventData.message_control); // or replace the generic parameter "object" with the right type for the argument you're using`
